opensoundscape/audio.py

```diff
@@ -469,6 +476,24 @@ def save(self, path):
 
         write(path, self.samples, self.sample_rate)
 
+        if write_metadata:
+            with soundfile.SoundFile(path, "r+") as s:
+                # must use r+ mode to update the file without overwriting everything
+                for field in [
```

Comment:
Why not iterate through all the fields in `self.metadata` and save them with the file?

Reply:
soundfile only supports this specific list of metadata fields, but an Audio object may have other fields. Trying to write other fields would raise an error

Reply:
I tested by writing a 'test' field with dummy value to the file after saving, can retrieve the value by `SoundFileObject.test` while it is open, but the same is not reflected in the metadata of the file when extracted through `TinyTag` or OpSo Audio object.